### PR TITLE
Load settings

### PR DESCRIPTION
**Pull request by s-scherrer**
Added a button to load the settings of an existing validation run into the validation run form.

### CODE REVIEW

**Comment by sheenaze** [comment]
Ok, after merging current master branch to this one there are a few issues:
1. When uploading settings from a published validation dates do not render properly. They do render properly if the validation is not published, and they don't if the validation is published;
2. When uploading settings and pressing 'validate' filters do not render properly (I'm checking this one right now);
3. There is "load settings" button on a result site, but not if the results are published, maybe it is a good idea to add it there as well?

**Comment by sheenaze** [comment]
One more thing, the problem about filters shows up only if more than one dataset is rendered on the left side.

**Comment by s-scherrer** [comment]
> One more thing, the problem about filters shows up only if more than one dataset is rendered on the left side.

The problem with the filters consists of two problems actually. The first problem happens when no initial filters are given, I added a check for this in the code and then at least only the correct filters for this dataset show up.

The second problem is because upon pressing 'validate', the page is rendered without passing the `valrun_uuid` parameter. Normally, clicking on "Load validation settings" loads `validate/?valrun_uuid=...`, and I infer the initial filters from the `ValidationRun` object. When this `valrun_uuid` is not passed, it assumes that this is the default validate page, and initial filters are only set for the default dataset (C3S) using the default initial filters.

So the issue should also show up when using only one dataset, but using a custom set of filters.

File: validator/views/validation.py [hunk 1]
```diff
@@ -207,12 +207,48 @@ def stop_validation(request, result_uuid):
...
+            {
+                "filters": dc.filters.all(),
+                "dataset": dc.dataset,
+                "version": dc.version,
+                "variable": dc.variable,
+            }
+            for dc in [ref_config] + data_configs
+        ]
+        ref_initial_values = initial_values[0]
+        data_initial_values = initial_values[1:]
```
Comment (s-scherrer) [inline validator/views/validation.py:240]: This is how I infer the initial filters to set.

Comment (s-scherrer) [reply validator/views/validation.py:240]: I think I could also infer the initial filters from `request.POST` for a POST request

**Comment by sheenaze** [comment]
Might be, but it shows up not only after reloading settings, also when you set more than one dataset and press validate it renders filters inapropriately.

**Comment by s-scherrer** [comment]
> Might be, but it shows up not only after reloading settings, also when you set more than one dataset and press validate it renders filters inapropriately.

Are you using the up to date version? I added a check that should at least render the filters correctly, but doesn't enable the ones that were set.

**Comment by sheenaze** [comment]
oh, ok, I'll update it.

**Comment by s-scherrer** [comment]
But I think it's possible to infer the correct filters from `request.POST.get`, I'm currently working on this.

**Comment by sheenaze** [comment]
Now it looks like it works :)

**Comment by sheenaze** [comment]
I mean, the rendering

**Comment by sheenaze** [comment]
Yea, rendering works now. And the button looks ok. So, only the issue of the date rendering when loading from the published validation left :)

**Comment by sheenaze** [comment]
Ok... sorry again for being a pain in the neck, but I realized it just now... what about ISMN network? It also should be loaded appropriately, otherwise it's not the same validation :(

**Comment by sheenaze** [comment]
And you spoiled my function for setting the appropriate time range :)

**Comment by s-scherrer** [comment]
> And you spoiled my function for setting the appropriate time range :)

I think it should now only be disabled when coming from another validation. I moved the check from your function to where the function is called.

**Comment by sheenaze** [comment]
I'll check it. I found a bug that I made and maybe I'll use your solution to fix it :D

**Comment by sheenaze** [comment]
Ok, I don't know if it is problem on my branch, maybe I have to update or downgrade my environment, but my problems are:
- dataset versions do not render properly after reloading,
- in case of ISMN networks do not render properly,
- triple collocation checkbox do not render properly (and I can not even change it) and it is interesting because it reads properly the scaling method and anomalies,
-  I can not open results run on the branch, it shows that there is a problem in qa4sm_reader:  val_ref = self.meta[globals._ref_ds_attr].
Would be good if someone else could check it as well, @wpreimes, @pstradio, could you take a look?

**Comment by sheenaze** [comment]
However, version rendering it's a tricky issue, it also doesn't render properly when looking for an existing validation, I have to fix it.

**Comment by sheenaze** [comment]
Tripple collocation issue as well :/

**Comment by pstradio** [comment]
> Ok, I don't know if it is problem on my branch, maybe I have to update or downgrade my environment, but my problems are:
> 
>     * dataset versions do not render properly after reloading,
> 
>     * in case of ISMN networks do not render properly,
> 
>     * triple collocation checkbox do not render properly (and I can not even change it) and it is interesting because it reads properly the scaling method and anomalies,
> 
>     * I can not open results run on the branch, it shows that there is a problem in qa4sm_reader:  val_ref = self.meta[globals._ref_ds_attr].
>       Would be good if someone else could check it as well, @wpreimes, @pstradio, could you take a look?

I had a look as well; what I found:

* The dataset versions issue doesn't seam to be present on my system
* I also were able to reproduce the problem with TC as the button to toggle 'include triple collocation..' is 'frozen' after reloading a validation (in which TC was used)
*  I also have a problem with the result files as the plots and netCDF **are not** generated in the run

As for the ISMN network issue, isn't it just because there is only one continent (wherever hawaii is placed) that includes all the data? (Therefore, no choice)

@sheenaze could you give some more info on opening the results on the branch? I don't quite understand what this refers to

**Comment by sheenaze** [comment]
I fetched your branch, so I worked on your code exactly, and when I started a validation being on this branch I was not able to get results. 
I managed to solve TC issue, so that the button is not frozen. It is enough to activate `setTcLock()` function anywhere in the code, I put it in line 1193, so it is next to the code to render dataset removal button. But it still doesn't render appropriately after refreshing.
Also, it doesn't make a difference what data you have available for your tests, chosen networks are saved in the database and they should be rendered appropriately. However, I have to check, because now I'm quite sure that it also doesn't work when you close the dialog window after it finds an existing validation.

**Comment by s-scherrer** [comment]
> I fetched your branch, so I worked on your code exactly, and when I started a validation being on this branch I was not able to get results.
> I managed to solve TC issue, so that the button is not frozen. It is enough to activate `setTcLock()` function anywhere in the code, I put it in line 1193, so it is next to the code to render dataset removal button. But it still doesn't render appropriately after refreshing.
> Also, it doesn't make a difference what data you have available for your tests, chosen networks are saved in the database and they should be rendered appropriately. However, I have to check, because now I'm quite sure that it also doesn't work when you close the dialog window after it finds an existing validation.

I get results, and versions are also rendered correctly. Did you reinstall your environment with the new qa4sm-reader version?

The ISMN problem is related to parametrised filters, I'll have to look into this.

Thanks for catching the TC issue and for solving it :) Can you tell me in which sense it's not rendering correctly?

**Comment by sheenaze** [comment]
I mean, even if I set triple collocation to be taken into account and then the dialog windows shows up and I close it, the checkbox is switched off. Ok, I have one idea how to solve it, but I'll check it tomorrow :)

**Comment by sheenaze** [comment]
Ok, TC works, but networks are not read properly :( 
Screenshot from 2021-02-25 08-39-22

On the right side you have settings, and on the left side you can see that none of the network is chosen :(

**Comment by sheenaze** [comment]
And when I run validation after reloading settings, I get sth like that:
Screenshot from 2021-02-25 08-42-35
when depth and networks filters are switched.

**Comment by s-scherrer** [comment]
> And when I run validation after reloading settings, I get sth like that:
> Screenshot from 2021-02-25 08-42-35
> when depth and networks filters are switched.

What's the issue here?

**Comment by pstradio** [comment]
I checked the latest commits and I couldn't reproduce your issue @sheenaze . To me it seems like all the filters are set correctly as in the previous validation. Could you possibly have an issue on your local setup?

**Comment by sheenaze** [comment]
Yeah, apparently it is on my machine. I have problems with tests as well, so now I'm trying to figure out what is wrong :) thx!

**Comment by sheenaze** [comment]
On my machine it switched both parameterised filters with each other. No idea why :(

**Comment by sheenaze** [comment]
I'm closing this pull request and I'll open a new one from the test_master branch. Samuel's changes are ok, but I found one small appearance issue and I want to fix it, so I'll add changes to the test master already.

**Pull request closed without merging**